EIPS/eip-6808.md

```diff
@@ -0,0 +1,680 @@
+) internal view returns (bool) {
+    TransferConditions memory conditions = _transferConditions[_account];
+
+    if (conditions.allFunds) {
+        return true;
+    }
+
+    if (
+        (conditions.amount == 0 &&
+            conditions.time == 0 &&
+            conditions.to == address(0) &&
+            !conditions.allFunds) ||
+        (conditions.amount > 0 && conditions.amount < _amount) ||
+        (conditions.time > 0 && conditions.time < block.timestamp) ||
+        (conditions.to != address(0) && conditions.to != _to)
+    ) {
+        return false;
+    }
+
+    return true;
+}
+```
+
+#### `_getAccountHolder` function
+
+This function identifies and returns the `_holder` account starting from the `sender` address.
+
+If the `sender` is not a `keyWallet` zero address is returned (`0x0`).
+
+```solidity
+function _getAccountHolder() internal view returns (address) {
+    address sender = _msgSender();
+    return
+        _firstAccounts[sender].accountHolderWallet != address(0)
+            ? _firstAccounts[sender].accountHolderWallet
+            : (
+                _secondAccounts[sender].accountHolderWallet != address(0)
+                    ? _secondAccounts[sender].accountHolderWallet
+                    : address(0)
+            );
+}
+```
+
+#### `_getOtherSecureWallet` function
+
+This function identifies and returns the other `keyWallet` starting from the `sender` address.
+
+If the `sender` is not a `keyWallet` zero address is returned (`0x0`).
+
+```solidity
+function _getOtherSecureWallet() internal view returns (address) {
+    address sender = _msgSender();
+    address accountHolder = _getAccountHolder();
+
+    return
+        _holderAccounts[accountHolder].firstWallet == sender
+            ? _holderAccounts[accountHolder].secondWallet
+            : _holderAccounts[accountHolder].firstWallet;
+}
+```
+
+#### `_isApprovalAllowed` function
+
+This function returns `true` if the `_time`, set in `allowApproval` function, has not elapsed yet.
+
+```solidity
+function _isApprovalAllowed(address account) internal view returns (bool) {
+    return _approvalConditions[account].time >= block.timestamp;
+}
+```
+
+#### `_afterTokenTransfer` function
+
+This function is called after a successful `transfer`, `transferFrom`, `mint` and `burn`.
+
+This function fires `Egress` event when loosing a holder and `Ingress` event when gaining a holder.
+
+```solidity
+function _afterTokenTransfer(
+    address from,
+    address to,
+    uint256 amount
+) internal virtual {
+    // region update secureAccounts
+    if (isSecureWallet(from) && balanceOf(from) == 0) {
+        delete _firstAccounts[_holderAccounts[from].firstWallet];
+        delete _secondAccounts[_holderAccounts[from].secondWallet];
+        delete _holderAccounts[from];
+    }
+    // endregion
+
+    if (balanceOf(from) == 0) {
+        emit Egress(from, amount);
+    }
+
+    if (balanceOf(to) == amount) {
+        emit Ingress(to, amount);
+    }
+}
+```
```

Comment:
```suggestion
```

Generally you should only standardize functions that are externally visible in the interface itself. The standard should provide enough information that multiple implementations can interoperate correctly, _without_ specifying how the implementations should work. These kind of choices should be left up to the implementers.